### PR TITLE
[KernelGen][Nvidia] Add cosine_similarity operator

### PR DESCRIPTION
**Pull request by ShawnsYing**
## Summary
Adds a Triton kernel for `cosine_similarity`. Computes the cosine similarity between two tensors along a specified dimension with broadcasting support.

## Testing
- Validated against reference on device via `to_reference(inp, True)`
- Tested on: Nvidia H20
- Test coverage: multiple dtypes (fp16/bf16/fp32/fp64), dimension reduction (dim=0/1/-1), broadcasting semantics, zero-vector eps protection, fp64 precision validation

## Performance
Test command: `pytest benchmark/test_cosine_similarity.py --level core` (NVIDIA H20)

### cosine_similarity

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|------|--------------------|-------------------|---------|
| float16 | [64, 64] | 0.034592 | 0.005824 | 5.940 |
| float16 | [256, 256] | 0.037344 | 0.005952 | 6.274 |
| float16 | [1024, 1024] | 0.056800 | 0.009120 | 6.228 |
| float16 | [4096, 4096] | 0.255328 | 0.029344 | 8.701 |
| float16 | [1024, 65536] | 0.897792 | 0.099200 | 9.050 |
| float32 | [64, 64] | 0.032256 | 0.005696 | 5.663 |
| float32 | [256, 256] | 0.035168 | 0.006304 | 5.579 |
| float32 | [1024, 1024] | 0.054928 | 0.011776 | 4.664 |
| float32 | [4096, 4096] | 0.312064 | 0.046848 | 6.661 |
| float32 | [1024, 65536] | 3.576656 | 0.158880 | 22.512 |
| bfloat16 | [64, 64] | 0.034688 | 0.005664 | 6.124 |
| bfloat16 | [256, 256] | 0.037536 | 0.006144 | 6.109 |
| bfloat16 | [1024, 1024] | 0.057184 | 0.009024 | 6.337 |
| bfloat16 | [4096, 4096] | 0.257952 | 0.029280 | 8.810 |
| bfloat16 | [1024, 65536] | 0.902464 | 0.101888 | 8.857 |

| Operator | Arithmetic Mean Speedup |
|----------|------------------------|
| cosine_similarity | **7.834** |

## Multi-backend Testing
| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---|---|
| Nvidia (H20) | PASS (15 cases) | 7.834 | Primary |
| Tianshu | N/A | — | — |
| Muxi | N/A | — | — |
| Ascend | N/A | — | — |
| Hygon | N/A | — | — |

## Files Changed
- `src/flag_gems/ops/cosine_similarity.py`: Triton kernel implementation with libtuner BLOCK_M/BLOCK_D tiling
- `tests/test_cosine_similarity.py`: Accuracy test suite
- `benchmark/test_cosine_similarity.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Reduction, stage: alpha 5.4, labels: aten/nn.functional/KernelGen)
- `src/flag_gems/runtime/backend/_nvidia/tune_configs.yaml`: Add 4 tuning configuration blocks

### CODE REVIEW

File: benchmark/test_cosine_similarity.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import pytest
```
Comment (bin913) [inline benchmark/test_cosine_similarity.py:15]: add copyright and license

File: tests/test_cosine_similarity.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import pytest
```
Comment (bin913) [inline tests/test_cosine_similarity.py:15]: add copyright and license

File: benchmark/test_cosine_similarity.py (hunk 1 above)
Comment (ShawnsYing) [reply benchmark/test_cosine_similarity.py:15]: Done! Added Apache 2.0 license header.

File: tests/test_cosine_similarity.py (hunk 2 above)
Comment (ShawnsYing) [reply tests/test_cosine_similarity.py:15]: Done! Added Apache 2.0 license header.

**Review by bin913 — DISMISSED** [review]

**Comment by bin913** [comment]
@ShawnsYing Please fix failing checks

**Review by bin913 — APPROVED** [review]
